aineko/extras/connectors/rest.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Extra module for connecting to a REST endpoint."""
+
+import json
+import time
+from typing import Any, Dict, Optional
+
+import requests
+from pydantic import BaseModel, field_validator
+
+from aineko import AbstractNode
+from aineko.extras.connectors.secrets import inject_secrets
+
+
+class ParamsREST(BaseModel):
+    """Connector params for REST model."""
+
+    timeout: int = 10
+    url: str
+    headers: Optional[Dict[str, Any]] = None
+    data: Optional[Dict[str, Any]] = None
+    poll_interval: int = 5
+    poll_throttle: float = 0.1
+    max_retries: int = 30
+    metadata: Optional[Dict[str, Any]] = None
```

Comment:
It's not clear how `metadata` is supposed to be used.

Reply:
Will document like in the WebSocketClient.